api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -140,15 +141,20 @@ protected User(final User source) {
         this.implMeta.putAll(source.implMeta);
     }
 
+    @Nullable
     public String getEmail() {
         return this.email;
     }
 
-    public void setEmail(final String email) {
-        this.setEmail(email, false);
+    public void setEmail(@Nullable final String email) {
```

Comment:
I originally made it NonNull, but there is a cascade effect from the point of getStringValue through the user mapper logic to here that was throwing a lint warning.

basically if the user we are loading from eDirectory is incomplete it could lead to a null email being returned. The other way to address this is to throw an Exception in that scenario.

It also raises the point of what should we initialize email to when initially creating the User object. `null` is handy in the sense that we haven't defined an email yet.